### PR TITLE
KAFKA-SITE DOCS 8227 - Add missing links Core Concepts duality of streams tables

### PR DESCRIPTION
**Pull request by londoncalling**
## Description

- Jira ticket: https://issues.apache.org/jira/browse/KAFKA-8227
- Related PR: https://github.com/apache/kafka/pull/6625
- add links from Core Concepts to relevant sections in Streams architecture and developer guides

## Review Requests
@bbejeck , @mjsax , @JimGalasyn , @joel-hamill 

Signed-off-by: Victoria Bialas <vicky@confluent.io>

### CODE REVIEW

File: 20/streams/core-concepts.html [hunk 1]
```diff
@@ -160,21 +161,26 @@ <h3><a id="streams_concepts_duality" href="#streams-concepts-duality">Duality of
...
         Any stream processing technology must therefore provide <strong>first-class support for streams and tables</strong>.
-        Kafka's Streams API provides such functionality through its core abstractions for
-        <code class="interpreted-text" data-role="ref">streams &lt;streams_concepts_kstream&gt;</code> and
-        <code class="interpreted-text" data-role="ref">tables &lt;streams_concepts_ktable&gt;</code>, which we will talk about in a minute.
-        Now, an interesting observation is that there is actually a <strong>close relationship between streams and tables</strong>,
-        the so-called stream-table duality.
-        And Kafka exploits this duality in many ways: for example, to make your applications
+        Kafka's Streams API provides such functionality through its core abstractions for 
+        <a id="streams_concepts_kstream" href="/{{version}}/documentation/streams/developer-guide/dsl-api#streams_concepts_kstream">streams</a>
+        <code class="interpreted-text" data-role="ref">&lt;streams_concepts_kstreams&gt;</code>
```
Comment (bbejeck) [inline 20/streams/core-concepts.html:166]: All the links point to the correct sections but I think we can get rid of all the `<code class="interpreted-text" data-role="ref">&lt;XX&gt;</code>` blocks in this section and elsewhere.  

I could be wrong, but It looked to me the `interpreted-text` sections were placeholders for setting the correct links pointing to other sections in the docs.

File: 21/streams/core-concepts.html [hunk 2]
```diff
@@ -160,21 +160,26 @@ <h3><a id="streams_concepts_duality" href="#streams-concepts-duality">Duality of
...
         Any stream processing technology must therefore provide <strong>first-class support for streams and tables</strong>.
-        Kafka's Streams API provides such functionality through its core abstractions for
-        <code class="interpreted-text" data-role="ref">streams &lt;streams_concepts_kstream&gt;</code> and
-        <code class="interpreted-text" data-role="ref">tables &lt;streams_concepts_ktable&gt;</code>, which we will talk about in a minute.
-        Now, an interesting observation is that there is actually a <strong>close relationship between streams and tables</strong>,
-        the so-called stream-table duality.
-        And Kafka exploits this duality in many ways: for example, to make your applications
+        Kafka's Streams API provides such functionality through its core abstractions for 
+        <a id="streams_concepts_kstream" href="/{{version}}/documentation/streams/developer-guide/dsl-api#streams_concepts_kstream">streams</a>
+        <code class="interpreted-text" data-role="ref">&lt;streams_concepts_kstreams&gt;</code>
```
Comment (bbejeck) [inline 21/streams/core-concepts.html]: same as above

File: 22/streams/core-concepts.html [hunk 3]
```diff
@@ -160,21 +160,26 @@ <h3><a id="streams_concepts_duality" href="#streams-concepts-duality">Duality of
...
         Any stream processing technology must therefore provide <strong>first-class support for streams and tables</strong>.
-        Kafka's Streams API provides such functionality through its core abstractions for
-        <code class="interpreted-text" data-role="ref">streams &lt;streams_concepts_kstream&gt;</code> and
-        <code class="interpreted-text" data-role="ref">tables &lt;streams_concepts_ktable&gt;</code>, which we will talk about in a minute.
-        Now, an interesting observation is that there is actually a <strong>close relationship between streams and tables</strong>,
-        the so-called stream-table duality.
-        And Kafka exploits this duality in many ways: for example, to make your applications
+        Kafka's Streams API provides such functionality through its core abstractions for 
+        <a id="streams_concepts_kstream" href="/{{version}}/documentation/streams/developer-guide/dsl-api#streams_concepts_kstream">streams</a>
+        <code class="interpreted-text" data-role="ref">&lt;streams_concepts_kstreams&gt;</code>
```
Comment (bbejeck) [inline 22/streams/core-concepts.html]: same as above

**Review by bbejeck — COMMENTED** [review]
@londoncalling thanks for the PR.  I just have some minor comments, but otherwise this PR LGTM.

File: 20/streams/core-concepts.html (hunk 1 above)
Comment (londoncalling) [reply 20/streams/core-concepts.html:166]: @bbejeck  Oh, I see, you just want linked terms only. Sure I can do that. (I assumed this was some API thing, but I guess it was suggestions for link targets.)

**Comment by londoncalling** [comment]
@mjsax , @joel-hamill , @JimGalasyn this should be good to go, can someone merge for me who has permissions?

**Review by bbejeck — APPROVED** [review]

**Comment by joel-hamill** [comment]
I spot checked a handful of the links locally and they looked good.

**Review by joel-hamill — APPROVED** [review]

**Comment by bbejeck** [comment]
Merged #204 into asf-site

**Comment by bbejeck** [comment]
Thanks for all the work @londoncalling!

**Pull request merged**